core/banking/erc20.go

```diff
@@ -135,7 +135,7 @@ func (e *Engine) DepositERC20(
 		logIndex:    logIndex,
 		txHash:      txHash,
 		chainID:     chainID,
-		bridgeView:  e.bridgeView,
+		bridgeView:  e.bridgeViewForChainID(chainID),
```

Comment:
Can we check that the `chainID` belongs to either the primary/second and error if not? I'm thinking of the case where a malicious validator sends in a hand-crafted ERC20 event with an incorrect chain-id. I think that would make core panic because the `assetAction` will be carrying around a `nil` bridge view.

Reply:
Sure.